wherescape/connectors/hubspot/process_data.py

```diff
@@ -0,0 +1,155 @@
+            hubspot_instance.send_contact_patch(inputs=properties)
+    if object_type == "deals":
+        if change_type == "patch":
+            hubspot_instance.send_deal_patch(inputs=properties)
+
+
+def create_data_dict(result: list, column_names: list, known_names: list):
+    """
+    Method to process a result list to a dict of keys id and properties.
+    All elements besides hubspot_company_id are stored in a dict under properties
+    The assumption is that the data per row is in the same order as the column names
+    """
+    result_dict = {}
+    property_dict = {}
+
+    for name in column_names:
+        data_item = result[column_names.index(name)]
+        if name in known_names:
+            """
+            For 1-to-1 column_names and property_names
+            """
+
+            if name == "hs_object_id":
+                """hs_object_id has to be an int"""
+                result_dict["id"] = int(data_item)
+            else:
+                if isinstance(data_item, Decimal):
+                    data_item = float(data_item)
+
+                property_dict[name] = data_item
+
+    result_dict.update({"properties": property_dict})
+    return result_dict
+
+
+def compare_names(source_names: list, destination_names: list):
+    """
+    This function compares source names with destiny names for one to one data transfer.
+    It will give a warning for source names that do not appear in the list of destiny names
+    """
+
+    known_destination_names = []
+
+    for name in source_names:
+        if name not in destination_names and name != "record_id":
+            logging.warning(
+                "source name: %s does not exist in the destination. please check its existence and spelling"
+                % name
+            )
+        else:
+            known_destination_names.append(name)
+
+    return known_destination_names
+
+
+def get_object_name(table_name: str):
+    """
+    This function willreturn the name of the object the data will be send to
+    """
+    if "companies" in table_name or "company" in table_name:
+        return "companies"
+    elif "contacts" in table_name or "contact" in table_name:
+        return "contacts"
+    elif "deals" in table_name or "deal" in table_name:
+        return "deals"
+    else:
+        logging.error(
+            "Could not identify the specific hubspot object type from the table name"
+        )
+
+
+def get_http_request_type(table_name: str):
+    """
+    This function will return the request_type based on the table name.
+    Currently, only
+    """
+    if "patch" in table_name:
+        return "patch"
+    else:
+        logging.error(
+            "Could not identify the specified API request desired from the table name"
+        )
+
+
+def get_property_names(object_name: str, hubspot_instance: Hubspot):
+    """
+    This function will return a list of propertynames of the selected object to compare to
+    """
+    if object_name == "companies":
```

Comment:
If we make get_companies_properties more generic this function can be simplified as well!
Then we just could do: return hubspot_instance.get_properties(object_name)